src/components/VideoPlayer2.tsx

```diff
@@ -29,335 +35,127 @@ const VOLUME_LEVELS: number[] = [0, 0.2, 0.4, 0.6, 0.8, 1.0];
 export const VideoPlayer: FunctionComponent<VideoPlayerProps> = ({
   setQuality,
   options,
+  videoList,
   contentId,
   onReady,
-  subtitles,
   onVideoEnd,
 }) => {
   const videoRef = useRef<HTMLDivElement>(null);
   const playerRef = useRef<Player | null>(null);
   const [player, setPlayer] = useState<any>(null);
+  const [currentVideoIndex, setCurrentVideoIndex] = useState<number>(0);
   const searchParams = useSearchParams();
-  useEffect(() => {
-    const t = searchParams.get('timestamp');
```

Comment:
why this diff?

Reply:
The behavior and difference in this code due to searchParams depend on how the rest of your application interacts with the URL and query parameters. For example, changing the value of timestamp in the URL may affect the behavior of the video player if that value is being used for seeking to a specific time in the video. This may lead to different behavior in the component if the value of timestamp changes or is manipulated in the URL.